modules/swagger/swaggerDiffImpl.ts

```diff
@@ -15,172 +17,183 @@ const LOCATION_MODEL = 'model';
 const LOCATION_PROPERTY = 'property';
 const LOCATION_PATH = 'path';
 
-/* TYPE EXTENSIONS */
 
-String.prototype.capitalizeFirstLetter = function() {
-	return this.charAt(0).toUpperCase() + this.slice(1);
-};
-
-Array.prototype.pushApply = function(arr) {
-	this.push.apply(this, arr);
-};
+declare global {
+	interface String {
+		capitalizeFirstLetter(): string;
+	}
 
-/* CONSTRUCTOR */
+	interface Array<T> {
+		pushApply(arr: T[]): void;
+	}
 
-function SwaggerDiffImpl() {
-	dot.templateSettings.strip = false;
 }
 
-/* PUBLIC PROPERTIES */
+String.prototype.capitalizeFirstLetter = function () {
+	return this.charAt(0).toUpperCase() + this.slice(1);
+};
 
-SwaggerDiffImpl.prototype.changes = {};
-SwaggerDiffImpl.prototype.changeCount = 0;
-SwaggerDiffImpl.prototype.swaggerInfo = {};
-// When [true], considers certain changes to be major changes instead of minor because they're breaking for SDKs
-SwaggerDiffImpl.prototype.useSdkVersioning = false;
-SwaggerDiffImpl.prototype.oldSwagger = {};
-SwaggerDiffImpl.prototype.newSwagger = {};
+Array.prototype.pushApply = function <T>(arr: T[]): void {
+	this.push(...arr);
+};
 
-/* PUBLIC FUNCTIONS */
+class SwaggerDiffImpl {
 
-SwaggerDiffImpl.prototype.diff = function(oldSwagger, newSwagger) {
-	console.log('Diffing swagger files...');
+	changes: { [key: string]: any } = {};
```

Comment:
no anyscript (there's a whole bunch in this file, not calling them out individually)